config/reflection_magic.go

```diff
@@ -76,7 +78,7 @@ func callConstructor(c reflect.Value, args []reflect.Value) (interface{}, error)
 	return val, err
 }
 
-type constructor func(h host.Host, u *tptu.Upgrader) interface{}
+type constructor func(h host.Host, u *tptu.Upgrader, cg connmgr.ConnectionGater) interface{}
```

Comment:
@aarshkshah1992 

What was the motivation to change this function signature? It's not really an issue, but it's redundant with the `ConnGater` field of the upgrader.

Reply:
Yes, I had the same thought but given the types are internal, I thought it wouldn't harm us to rectify later. Wanna tackle that as a follow-up, @aarshkshah1992?

Reply:
@Stebalien @raulk 

I think this was an oversight. I'll create a follow up PR to get rid of this. #951 .

Reply:
Thanks! Note: this isn't a super high priority.